# Yair Motro, PhD
<yair.motro@gmail.com> | [motroy.github.io/blog](https://motroy.github.io/blog)

---

### **WORK EXPERIENCE**

**Bioinformatics Lead**
_Jan 2016 - Present_
_Microbiology, Advanced Genomics and Infection Control Applications Laboratory, Ben Gurion University of the Negev_

As bioinformatics lead, I am responsible for all computing requirements of the lab, including research and technical applications. For research, I am involved in a variety of projects and collaborations covering the three major branches of microbial Next-Generation Sequencing (NGS): shotgun metagenomics, targeted metagenomics, and whole genome sequencing. For technical aspects, I manage the storage of lab data, computers, and workstations used by lab members.

---

### **PROJECTS**

**Whole Genome Sequencing (WGS) based Genomic Surveillance of Clinical Pathogens**

Involved in the curation and analysis of WGS of pathogenic bacterial clinical isolates as part of a genomic-focused hospital outbreak surveillance project. Data curation involved both the organization of contextual data and the deposition of WGS sequence data. Analyses involved applying computational tools and pipelines for species ID, genome annotation (including the resistome, virulome, and plasmidome), phylogenomic comparisons (including SNPs and cgMLST), and compiling actionable summary reports for decision-makers, assisting in attempts to prevent and manage outbreaks in hospitals.

**WGS in One Health Context**

Involved in analyses of WGS data of isolates from different contexts (for example, environments including wastewater and livestock) to determine genomic relatedness to support or reject epidemiologically based hypotheses.

**Targeted Microbiome-based Research**

Involved in curating and analyzing NGS data from targeted metagenomic sequencing projects, covering a diverse range of environments (including different human body sites, animals, and water) and research questions (including clinical, veterinary, and public health-related implications). Analyses involved applying targeted metagenomic sequence analysis pipelines (such as QIIME2 and Mothur) and downstream analyses using publicly available R packages and Python scripts. Much of these analyses were done with continual collaboration with domain-specific experts.

**Untargeted Metagenomics-based Research**

Involved in curating and analyzing NGS data from untargeted/shotgun metagenomic sequencing projects, once again covering a diverse range of environments and research questions. Applying state-of-the-art methods for analysis, including pipeline development. Downstream analyses involve the use of Jupyter and Rmarkdown notebooks with a range of cloud environment setups.

**Microbial Bioinformatics Tools and Pipelines**

Involved in the development and implementation of a number of tools and pipelines for microbial NGS analyses. Pipelines include analyses involving pathogen detection and isolation from shotgun metagenome samples (including Working dog and Golden Jackal faeces, freshwater, wastewater, and more), phylogenomic and resistome analysis of WGS from environmental and clinical isolates, microbial gWAS and pWAS analyses, plasmids-focused analyses, web-based applications for NGS data simulation tools and WGS-based pipelines, genome editing pipelines, and more. Also worked in collaboration with BGU School of Computer Science in the development of datasets for adversarial attacks and machine learning.

---

### **EDUCATION**

*   **Post-Doc (Theoretical Chemistry)**, Ben Gurion University of the Negev (BeerSheva, Israel), _2014 - 2016_
*   **PhD (Bioinformatics)**, Murdoch University (Perth, Western Australia), _2004 - 2008_
*   **BSc in Biotechnology**, Murdoch University (Perth, Western Australia), _2000 - 2003_
*   **BSc in Computer Science**, Murdoch University (Perth, Western Australia), _2000 - 2003_

---

### **SKILLS**

*   **Bioinformatics Analysis**: Whole Genome Sequencing (WGS), Targeted & Untargeted Metagenomics, Microbiome Analysis (QIIME2, Mothur), Phylogenomics (SNP & cgMLST-based), Resistome & Virulome Analysis, Microbial GWAS/pWAS, Plasmid Analysis.
*   **Programming & DevOps**:
    *   **Languages**: Python, R, Bash
    *   **Tools**: Jupyter Notebooks, RMarkdown, Git, GitHub, GitLab, VScode
    *   **LLM Tools**: prompt engineering with gemini and agenic programming with jules
    *   **Pipelines**: Snakemake, Nextflow
    *   **Infrastructure**: HPC cluster management, Cloud computing environments, Docker, Singularity.
*   **Data Science & Machine Learning**: Development of datasets for machine learning, analysis of adversarial attacks on NGS data, application of ML for strain typing (minMLST).
*   **Project Management & Communication**: Managing multiple concurrent projects with diverse collaborators, authoring scientific publications and technical documentation, presenting at international conferences (e.g., ECCMID).

---

### **PUBLICATIONS**

**2025**

*   Bardenstein S, Grupel D, Even-Tov B, Motro Y, Moran-Gilad J. (2025). "Reemergence of Brucella abortus, Israel, 2021." _Emerging Infectious Diseases_, 31(4), pp. 820-823.
*   Craddock HA, Resnick KA, Gorovsky O, Salah I, Abu Ramileh M, Nesher L, Yagel Y, Sagi O, Troib S, Motro Y, Moran-Gilad J. (2025). "Tracking Extended-Spectrum Beta Lactamase-Producing Enterobacterales Across One Health Among Bedouin Communities." _Open Forum Infectious Diseases_, ofaf495.
*   Kraus O, Schwarz Y, Motro Y, Marom T, Lamdan R, Kapuller V, Ovnat ST, Moran-Gilad J. (2025). "Adenotonsillar Microbiome Shifts in Children With Obstructive Sleep Apnea Syndrome." _The Laryngoscope_.
*   Landman D, Cherniak M, Gefen O, Michael-Gayego A, Motro Y, Moran-Gilad J, Strahilevitz J. (2025). "Bactericidal activity of antibiotic combinations against clinical isolates of Enterococcus gallinarum and Enterococcus casseliflavus." _Journal of Antimicrobial Chemotherapy_, pp. 1519-1525.
*   Lapid R, Motro Y, Craddock H, Salah I, King R, Winner K, Kahila Bar-Gal G, Moran-Gilad J. (2025). "Abundance of clinically relevant antimicrobial resistance genes in the golden jackal (Canis aureus) gut." _mSphere_, e0081924.
*   Lapid R, Motro Y, Juravel K, King R, Moran-Gilad J, Kahila Bar-Gal G. (2025). "The golden jackal (Canis aureus): interaction of fur and fecal microbiota, host genetics and animal traits." _Animal Microbiome_, 106.
*   Marano RBM, Oster Y, Benenson S, Motro Y, Ayalon O, Rosenbluh C, Cuénod A, Michael-Gayego A, Temper V, Strahilevitz J, Moran-Gilad J. (2025). "An Omics-Guided Investigation of a Hospital Outbreak Caused by blaNDM-1-Producing Pseudocitrobacter faecalis." _The Journal of Infectious Diseases_, pp. e17-e26.
*   Yagel Y, Motro Y, Green S, Klapper-Goldstein H, Pardo E, Moran-Gilad J, Weintraub AY. (2025). "Investigation of the female genital tract microbiome and its association with hydrosalpinx in women undergoing salpingectomy." _Archives of Gynecology and Obstetrics_, pp. 1649-1656.

**2024**

*   Motro Y, Temper V, Strahilevitz J, Moran-Gilad J. (2024). "Invasive infections caused by the recently described species Enterococcus innesii." _European Journal of Clinical Microbiology & Infectious Diseases_, pp. 1645-1650.
*   Oren I, Temper V, Michael-Gayego A, Motro Y, Volovelsky O, Moran-Gilad J, Gordon O. (2024). "Helicobacter cinaedi Bacteremia in Children: A Case Report and Literature Review." _The Pediatric Infectious Disease Journal_, pp. e135-e138.
*   Posteraro B, De Maio F, Motro Y, Menchinelli G, De Lorenzis D, Marano RBM, Aljanazreh B, Errico FM, Massaria G, Spanu T, Posteraro P, Moran-Gilad J, Sanguinetti M. (2024). "In-depth characterization of multidrug-resistant NDM-1 and KPC-3 co-producing Klebsiella pneumoniae bloodstream isolates from Italian hospital patients." _Microbiology Spectrum_, e0330523.
*   Strahilevitz J, Motro Y, Temper V, Merezhko D, Ayalon O, Bar Moshe Y, Lam MMC, Holt KE, Moran-Gilad J. (2024). "In vivo selection of carbapenem resistance during persistent Klebsiella pneumoniae sequence type 395 bloodstream infection due to OmpK36 deletion." _Antimicrobial Agents and Chemotherapy_, e00663-24.
*   Yagel Y, Motro Y, Green S, Klapper-Goldstein H, Pardo E, Moran-Gilad J, Weintraub AY. (2024). "Investigation of the Female Genital Tract Microbiome and its Association with Hydrosalpinx in Women Undergoing Salpingectomy." _Research Square_, (Preprint).

**2023**

*   Bardenstein S, Grupel D, Blum SE, Motro Y, Moran-Gilad J. (2023). "Public and animal health risks associated with spillover of Brucella melitensis into dairy farms." _Microbial Genomics_.
*   Grupel D, Sagi O, Nissan I, Grossman R, Yair M, Moran-Gilad J, Danino D. (2023). "Challenging diagnosis of Mycolicibacterium cosmeticum/canariasense infection: A case report and literature review." _Journal of Clinical Tuberculosis and Other Mycobacterial Diseases_, 100393.
*   Horev A, Brandwein M, Vaknin A, Motro Y, Moran-Gilad J. (2023). "Temporal Changes in the Skin Microbiome of Epidermolysis Bullosa Patients following the Application of Wound Dressings." _Journal of Clinical Medicine_, 12(20), p. 6435.
*   Lapid R, Motro Y, Craddock H, Khalfin B, King R, Bar-Gal GK, Moran-Gilad J. (2023). "Fecal microbiota of the synanthropic golden jackal (Canis aureus)." _Animal Microbiome_, 37.
*   Meiseles A, Motro Y, Rokach L, Moran-Gilad J. (2023). "Vulnerability of pangolin SARS-CoV-2 lineage assignment to adversarial attack." _Artificial Intelligence in Medicine_, 102722.
*   Motro Y, Wajnsztajn D, Michael-Gayego A, Mathur S, Marano RB, Salah I, Rosenbluh C, Temper V, Strahilevitz J, Moran-Gilad J. (2023). "Metagenomic sequencing for investigation of a national keratoconjunctivitis outbreak, Israel, 2022." _Eurosurveillance_.
*   Zorea J, Motro Y, Mazor RD, Carmi YK, Shulman Z, Mahajna J, Moran-Gilad J, Elkabets M. (2023). "TRAF3 Suppression Encourages B Cell Recruitment and Prolongs Survival of Microbiome-Intact Mice with Ovarian Cancer." _Journal of Experimental & Clinical Cancer Research_, 42(1), p. 107.

**2022**

*   Cohen A, Poupko L, Craddock HA, Motro Y, Khalfin B, Zelinger A, Tirosh-Levy S, Blum SE, Steinman A, Moran-Gilad J. (2022). "Fecal Microbiome Features Associated with Extended-Spectrum β-Lactamase-Producing Enterobacterales Carriage in Dairy Heifers." _Animals_, 12(14), p. 1738.
*   Craddock HA, Motro Y, Zilberman B, Khalfin B, Bardenstein S, Moran-Gilad J. (2022). "Long-Read Sequencing and Hybrid Assembly for Genomic Analysis of Clinical Brucella melitensis Isolates." _Microorganisms_, 10(3), p. 619.
*   Craddock HA, Godneva A, Rothschild D, Motro Y, Grinstein D, Lotem-Michaeli Y, Narkiss T, Segal E, Moran-Gilad J. (2022). "Phenotypic correlates of the working dog microbiome." _npj Biofilms and Microbiomes_, 8(1), p. 66.
*   Kaniel O, Sherf-Dagan S, Szold A, Langer P, Khalfin B, Kessler Y, Raziel A, Sakran N, Motro Y, Goitein D, Moran-Gilad J. (2022). "The Effects of One Anastomosis Gastric Bypass Surgery on the Gastrointestinal Tract." _Nutrients_, 14(2), p. 304.
*   Lazarovits G, Gefen O, Cahanian N, Adler K, Fluss R, Levin-Reisman I, Ronin I, Motro Y, Moran-Gilad J, Balaban NQ, Strahilevitz J. (2022). "Prevalence of Antibiotic Tolerance and Risk for Reinfection Among Escherichia coli Bloodstream Isolates: A Prospective Cohort Study." _Clinical Infectious Diseases_, pp. 1706-1713.
*   Mendes CI, Vila-Cerqueira P, Motro Y, Moran-Gilad J, Carriço JA, Ramirez M. (2022). "LMAS: evaluating metagenomic short de novo assembly methods through defined communities." _GigaScience_, giac122.
*   Scaturro M, Chierico FD, Motro Y, et al. (2022). "Bacterial communities of premise plumbing systems in four European cities, and their association with culturable Legionella." _bioRxiv_, (Preprint).
*   Sofer-Sali N, Roif-Kaminsky D, Motro Y, Khalfin B, Avramovich E, Galor I, Shlaifer A, Sommer A, Rutenberg R, Sachter Y, Yitzhak A, Grupel D, Moran-Gilad J. (2022). "Prevalence and Characteristics of Carriage of Neisseria meningitidis Among Young Israeli Adults." _Open Forum Infectious Diseases_, ofac482.
*   Zilberman B, Motro Y, Sagi O, Kornspan D, Ben-Shimol S, Gdalevich M, Yagel Y, Davidovitch N, Khalfin B, Rabinowitz P, Nesher L, Grotto I, Bardenstein S, Moran-Gilad J. (2022). "Genomic Epidemiology of Clinical Brucella melitensis Isolates from Southern Israel." _Microorganisms_, 10(2), p. 238.

**2021**

*   Bardenstein S, Gibbs RE, Yagel Y, Motro Y, Moran-Gilad J. (2021). "Brucellosis Outbreak Traced to Commercially Sold Camel Milk through Whole-Genome Sequencing, Israel." _Emerging Infectious Diseases_, pp. 1728-1731.
*   Cohen S, Rokach L, Motro Y, Moran-Gilad J, Veksler-Lublinsky I. (2021). "minMLST: machine learning for optimization of bacterial strain typing." _Bioinformatics_, pp. 303-311.
*   Domanovich-Asor T, Craddock HA, Motro Y, Khalfin B, Peretz A, Moran-Gilad J. (2021). "Unraveling antimicrobial resistance in Helicobacter pylori: Global resistome meets global phylogeny." _Helicobacter_, e12782.
*   O'Sullivan DM, Doyle RM, Temisak S, et al. (2021). "An inter-laboratory study to investigate the impact of the bioinformatics component on microbiome analysis using mock communities." _Scientific Reports_, 10590.

**2020**

*   Bouganim R, Dykman L, Fakeh O, Motro Y, Oren R, Daniel C, Lazarovitch T, Zaidenstein R, Moran-Gilad J, Marchaim D. (2020). "The Clinical and Molecular Epidemiology of Noncarbapenemase-Producing Carbapenem-Resistant Enterobacteriaceae: A Case-Case-Control Matched Analysis." _Open Forum Infectious Diseases_, ofaa299.
*   Domanovich-Asor T, Motro Y, Khalfin B, Craddock HA, Peretz A, Moran-Gilad J. (2020). "Genomic Analysis of Antimicrobial Resistance Genotype-to-Phenotype Agreement in Helicobacter pylori." _Microorganisms_, E2.
*   Doyle RM, O'Sullivan DM, Aller SD, et al. (2020). "Discordant bioinformatic predictions of antimicrobial resistance from whole-genome sequencing data of bacterial isolates: an inter-laboratory study." _Microbial Genomics_.
*   Fayans I, Motro Y, Rokach L, Oren Y, Moran-Gilad J. (2020). "Cyber security threats in the microbial genomics era: implications for public health." _Eurosurveillance_.
*   Yagel Y, Sestito S, Motro Y, Shnaiderman-Torban A, Khalfin B, Sagi O, Navon-Venezia S, Steinman A, Moran-Gilad J. (2020). "Genomic Characterization of Antimicrobial Resistance, Virulence, and Phylogeny of the Genus Ochrobactrum." _Antibiotics_, E177.

**2019**

*   Doyle RM, O’Sullivan DM, Aller SD, et al. (2019). "Discordant bioinformatic predictions of antimicrobial resistance from whole-genome sequencing data of bacterial isolates: An inter-laboratory study." _bioRxiv_, (Preprint).

**2018**

*   Motro Y, Carriço JA, Friedrich AW, Rossen JWA, Moran-Gilad J. (2018). "ESCMID postgraduate education course: regional capacity building for integration of next-generation sequencing in the clinical microlab." _Microbes and Infection_, pp. 275-280.

**2017**

*   Motro Y, Moran-Gilad J. (2017). "Next-generation sequencing applications in clinical bacteriology." _Biomolecular Detection and Quantification_, pp. 1-6.
*   Nanda J, Rubinov B, Ivnitski D, et al. (2017). "Emergence of native peptide sequences in prebiotic replication networks." _Nature Communications_, 434.

**2016**

*   Daschakraborty S, Kiefer PM, Miller Y, Motro Y, Pines D, Pines E, Hynes JT. (2016). "Reaction Mechanism for Direct Proton Transfer from Carbonic Acid to a Strong Base in Aqueous Solution I: Acid and Base Coordinate and Charge Dynamics." _The Journal of Physical Chemistry B_, pp. 2271-2280.
*   Daschakraborty S, Kiefer PM, Miller Y, Motro Y, Pines D, Pines E, Hynes JT. (2016). "Reaction Mechanism for Direct Proton Transfer from Carbonic Acid to a Strong Base in Aqueous Solution II: Solvent Coordinate-Dependent Reaction Path." _The Journal of Physical Chemistry B_, pp. 2281-2290.

**2014**

*   Linnewiel-Hermoni K, Motro Y, Miller Y, Levy J, Sharoni Y. (2014). "Carotenoid derivatives inhibit nuclear factor kappa B activity in bone and cancer cells by targeting key thiol groups." _Free Radical Biology and Medicine_, pp. 105-120.

**2009**

*   Bellgard MI, Wanchanthuek P, La T, et al. (2009). "Genome sequence of the pathogenic intestinal spirochete brachyspira hyodysenteriae reveals adaptations to its lifestyle in the porcine large intestine." _PLoS One_, e4641.
*   Motro Y, La T, Bellgard MI, Dunn DS, Phillips ND, Hampson DJ. (2009). "Identification of genes associated with prophage-like gene transfer agents in the pathogenic intestinal spirochaetes Brachyspira hyodysenteriae, Brachyspira pilosicoli and Brachyspira intermedia." _Veterinary Microbiology_, pp. 340-345.

**2006**

*   Binnewies TT, Motro Y, Hallin PF, et al. (2006). "Ten years of bacterial genome sequencing: comparative-genomics-based discoveries." _Functional & Integrative Genomics_, pp. 165-185.
