{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1ffd9a04-6b4b-4655-bfe9-d88bed0555ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./\n",
      "./index-for-model-sequences.txt.gz\n",
      "./card-genomes.txt.gz\n",
      "./61_kmer_db.json.gz\n",
      "./nucleotide_fasta_protein_overexpression_model_variants.fasta.gz\n",
      "./protein_fasta_protein_variant_model_variants.fasta.gz\n",
      "./nucleotide_fasta_protein_homolog_model_variants.fasta.gz\n",
      "./card_prevalence.txt.gz\n",
      "./all_amr_61mers.txt.gz\n",
      "./nucleotide_fasta_rRNA_gene_variant_model_variants.fasta.gz\n",
      "./nucleotide_fasta_protein_variant_model_variants.fasta.gz\n",
      "./Variants-Download-README.txt\n",
      "./protein_fasta_protein_overexpression_model_variants.fasta.gz\n",
      "./protein_fasta_protein_homolog_model_variants.fasta.gz\n"
     ]
    }
   ],
   "source": [
    "#!wget \"https://card.mcmaster.ca/download/6/prevalence-v4.0.0.tar.bz2\"\n",
    "#!tar xvf prevalence-v4.0.0.tar.bz2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "19d097e0-a973-4dcf-8d52-4848124c0c19",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gunzip card-genomes.txt.gz index-for-model-sequences.txt.gz card_prevalence.txt.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "df65514b-d253-48ed-b552-ac31f9dd9109",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CARD Variants Download README\n",
      "\n",
      "Use or reproduction of these materials, in whole or in part, by any commercial \n",
      "organization whether or not for non-commercial (including research) or commercial purposes\n",
      "is prohibited, except with written permission of McMaster University. Commercial uses are\n",
      "offered only pursuant to a written license and user fee. To obtain permission and begin \n",
      "the licensing process, see http://card.mcmaster.ca/about.\n",
      "\n",
      "For details on how these data are generated, see https://card.mcmaster.ca/genomes and\n",
      "https://card.mcmaster.ca/prevalence.\n",
      "\n",
      "CITATION:\n",
      "\n",
      "Alcock et al. \"CARD 2020: antibiotic resistome surveillance with the Comprehensive\n",
      "Antibiotic Resistance Database\" Nucleic Acids Research, 48: D517-D525.\n",
      "https://www.ncbi.nlm.nih.gov/pubmed/31665441\n",
      "\n",
      "CARD SHORT NAMES:\n",
      "\n",
      "A CARD-specific abbreviation for AMR gene names associated with Antibiotic Resistance\n",
      "Ontology terms, often not based on the literature. This is used for programmatic and \n",
      "compatibility purposes and is not ontologically relevant. Each ontology term with an \n",
      "associated AMR detection model has a CARD Short Name that appears in CARD data files \n",
      "and output generated by RGI. If the original gene name is less than 15 characters, the \n",
      "CARD short name is identical; if the gene name is greater than 15 characters, the CARD \n",
      "Short Name has been abbreviated by CARD curators specifically to identify the proper \n",
      "gene or protein name. All CARD Short Names are unique and have whitespace characters \n",
      "replaced by underscore characters. The convention for pathogen names is capitalized \n",
      "first letter of the genus followed by the lowercase first three letters of the species \n",
      "name. The antibiotic abbreviations are from https://journals.asm.org/journal/aac/abbreviations\n",
      "plus some custom abbreviations by the CARD curators. Simple CARD Short Names often do not\n",
      "involve either, e.g. CTX-M-15, but where applicable the CARD Short Names follow pathogen_gene\n",
      "or pathogen_gene_drug. The full lists of abbreviations can be found at:\n",
      "https://card.mcmaster.ca/latest/data\n",
      "\n",
      "\n",
      "FASTA:\n",
      "\n",
      "The provided FASTA files are separated for both sequence type (nucleotide and peptide sequences)\n",
      "and for model type. The included AMR detection model types from CARD are: protein homolog model, \n",
      "protein variant model, protein overexpression model and rRNA gene variant model. Note that the\n",
      "rRNA gene variant model is nucleotide-only as this model identifies putative AMR-conferring \n",
      "mutations in rRNA. For an in-depth description of each model type, please see the CARD website or\n",
      "Alcock et al., 2020. The header for each FASTA sequence includes: 1) the prevalence \n",
      "sequence ID (crossreferenced with the \"index-for-model-sequences.txt\" described below); 2) the ARO \n",
      "name for that determninant as it appears in CARD; 3) the ARO accession for that determinant; \n",
      "and, 4) the detection model type used to predict that determinant. Please note these sequences \n",
      "are as they appeared from assembly and so do not directly indicate the predicted AMR-relevant \n",
      "mutation (as applicable to protein variant, protein overexpression and rRNA gene variant models).\n",
      "In addition, in a small number of cases sequences may have multiple ARO names and accessions\n",
      "delimited by commas.\n",
      "\n",
      "\n",
      "INDEX FILES:\n",
      "\n",
      "\"index-for-model-sequences.txt.gz\": contains all the detection statistics for the \n",
      "sequences available in the above FASTA files, indicating pathogen, detection criteria, \n",
      "ARO categorization, and similarity to curated CARD reference sequence.\n",
      "\n",
      "The headers included are: \n",
      "\n",
      "\"prevalence_sequence_id\": indicates the corresponding sequence in the FASTA file\n",
      "\"model_id\": CARD-assigned ID indicating the AMR detection model\n",
      "\"aro_term\": indicates the name of the AMR determinant to which the model is attached\n",
      "\"aro_accession\": the unique ARO accession for each ontology term in CARD\n",
      "\"detection_model\": the detection model type used for this determinant\n",
      "\"species_name\": the analyzed pathogen in which this determinant was detected\n",
      "\"ncbi_accession\": the NCBI RefSeq accession from which this determinant was detected\n",
      "\"data_type\": the assembly level of the analyzed genome (chromosome, plasmid, wgs)\n",
      "\"rgi_criteria\": the confidence of the AMR determinant hit from RGI (Perfect or Strict)\n",
      "\"percent_identity\": the percent identity of the matched sequence region to the CARD reference\n",
      "\"bitscore\": the bitscore of the matched sequence region to the CARD reference\n",
      "\"amr_gene_family\": the AMR Gene Family from CARD for this determinant\n",
      "\"resistance_mechanism\": the Resistance Mechanism from CARD for this determinant\n",
      "\"drug_class\": the Drug Class from CARD for this determinant\n",
      "\n",
      "\n",
      "\"card-genomes.txt.gz\": tab-separated; indicates for each analyzed genome which putative AMR \n",
      "determinants were detected by RGI at both the Perfect and Strict confidence levels. This \n",
      "provides a \"putative resistome\"  for each genome. Available and viewable on the web at:\n",
      " https://card.mcmaster.ca/genomes.\n",
      "\n",
      "The headers included are: \n",
      "\n",
      "\"dna_accession\": the NCBI RefSeq accession analyzed \n",
      "\"pathogen\": the pathogen/species indicated for this assembly \n",
      "\"data_source\": the assembly level of the analyzed genome (chromosome, plasmid, wgs)\n",
      "\"perfect_hits\": comma-separated list of AMR determinants identified as \"perfect\" by RGI\n",
      "\"strict_hits\": comma-separated list of AMR determinants identified as \"strict\" by RGI\n",
      "\n",
      "\n",
      "\"card_prevalence.txt.gz\": tab-separated; indicates the prevalence of each detected AMR \n",
      "determinant by  assembly type (chromosome, plasmid, wgs) and RGI confidence crtieria \n",
      "(Perfect or Strict). Available  and viewable on the web at https://card.mcmaster.ca/prevalence.\n",
      "\n",
      "The headers included are:\n",
      "\n",
      "\"ARO Accession\": the unique ARO identifier from CARD for each ontology term\n",
      "\"Name\": the name for this accession as it appears in CARD\n",
      "\"Model ID\": the AMR detection model ID used to predict this determinant\n",
      "\"Model Type\": the AMR detection model type used to predict this determinant\n",
      "\"Pathogen\": the pathogen/species being described by prevalence statistics\n",
      "\"NCBI Plasmid / NCBI Chromosome / NCBI WGS / NCBI Genomic Island\": the prevalence (as %) of \n",
      "this determinant across all analyzed assemblies for this data type \n",
      "\"Criteria\": the RGI criteria (perfect or strict) used to calculate the prevalence of this \n",
      "determinant\n",
      "\"ARO Categories\": semi-colon-separated list of ARO categories listed for this determinant \n",
      "(AMR Gene Family, Resistance Mechanism, Drug Class)\n",
      "\n",
      "KMER FILES:\n",
      "\n",
      "The files \"61_kmer_db.json.gz\" and \"all_amr_61mers.txt.gz\" are used by the Resistome Gene\n",
      "Identifier (RGI) for pathogen-of-origin analysis of predicted antimicrobial resistance\n",
      "genes, see https://github.com/arpcard/rgi.\n"
     ]
    }
   ],
   "source": [
    "!cat Variants-Download-README.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "869f434f-a7df-4d3e-b9f1-86bff7232370",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!mamba install -y pandas\n",
    "import pandas as pd\n",
    "\n",
    "prevalence = pd.read_csv(\"card_prevalence.txt\", sep=\"\\t\")\n",
    "genomes = pd.read_csv(\"card-genomes.txt\", sep=\"\\t\")\n",
    "index_model = pd.read_csv(\"index-for-model-sequences.txt\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "e76a8040-1b64-46a5-b5ad-a2dc68a814d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ARO Accession</th>\n",
       "      <th>Name</th>\n",
       "      <th>Model ID</th>\n",
       "      <th>Model Type</th>\n",
       "      <th>Pathogen</th>\n",
       "      <th>NCBI Plasmid</th>\n",
       "      <th>NCBI WGS</th>\n",
       "      <th>NCBI Chromosome</th>\n",
       "      <th>NCBI Genomic Island</th>\n",
       "      <th>Criteria</th>\n",
       "      <th>ARO Categories</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ARO:3001109</td>\n",
       "      <td>SHV-52</td>\n",
       "      <td>4</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ARO:3001109</td>\n",
       "      <td>SHV-52</td>\n",
       "      <td>4</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>ARO:3001989</td>\n",
       "      <td>CTX-M-130</td>\n",
       "      <td>7</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; cephalosporin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>ARO:3002356</td>\n",
       "      <td>NDM-6</td>\n",
       "      <td>8</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>ARO:3002356</td>\n",
       "      <td>NDM-6</td>\n",
       "      <td>8</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21929</th>\n",
       "      <td>ARO:3007087</td>\n",
       "      <td>KPC-90</td>\n",
       "      <td>5770</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21932</th>\n",
       "      <td>ARO:3007089</td>\n",
       "      <td>KPC-83</td>\n",
       "      <td>5771</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21940</th>\n",
       "      <td>ARO:3007102</td>\n",
       "      <td>KPC-93</td>\n",
       "      <td>5776</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21941</th>\n",
       "      <td>ARO:3007102</td>\n",
       "      <td>KPC-93</td>\n",
       "      <td>5776</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21972</th>\n",
       "      <td>ARO:3007120</td>\n",
       "      <td>tet(O/M/O)</td>\n",
       "      <td>5790</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic target protection; tetracycline ant...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1634 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      ARO Accession        Name  Model ID             Model Type  \\\n",
       "4       ARO:3001109      SHV-52         4  protein homolog model   \n",
       "5       ARO:3001109      SHV-52         4  protein homolog model   \n",
       "40      ARO:3001989   CTX-M-130         7  protein homolog model   \n",
       "45      ARO:3002356       NDM-6         8  protein homolog model   \n",
       "46      ARO:3002356       NDM-6         8  protein homolog model   \n",
       "...             ...         ...       ...                    ...   \n",
       "21929   ARO:3007087      KPC-90      5770  protein homolog model   \n",
       "21932   ARO:3007089      KPC-83      5771  protein homolog model   \n",
       "21940   ARO:3007102      KPC-93      5776  protein homolog model   \n",
       "21941   ARO:3007102      KPC-93      5776  protein homolog model   \n",
       "21972   ARO:3007120  tet(O/M/O)      5790  protein homolog model   \n",
       "\n",
       "                    Pathogen  NCBI Plasmid  NCBI WGS  NCBI Chromosome  \\\n",
       "4      Klebsiella pneumoniae          0.00      0.11              0.0   \n",
       "5      Klebsiella pneumoniae          0.00      0.16              0.0   \n",
       "40     Klebsiella pneumoniae          0.00      0.02              0.0   \n",
       "45     Klebsiella pneumoniae          0.01      0.01              0.0   \n",
       "46     Klebsiella pneumoniae          0.01      0.01              0.0   \n",
       "...                      ...           ...       ...              ...   \n",
       "21929  Klebsiella pneumoniae          0.00      0.02              0.0   \n",
       "21932  Klebsiella pneumoniae          0.03      0.06              0.0   \n",
       "21940  Klebsiella pneumoniae          0.03      0.01              0.0   \n",
       "21941  Klebsiella pneumoniae          0.03      0.00              0.0   \n",
       "21972  Klebsiella pneumoniae          0.00      0.02              0.0   \n",
       "\n",
       "       NCBI Genomic Island        Criteria  \\\n",
       "4                        0         perfect   \n",
       "5                        0  perfect_strict   \n",
       "40                       0  perfect_strict   \n",
       "45                       0  perfect_strict   \n",
       "46                       0         perfect   \n",
       "...                    ...             ...   \n",
       "21929                    0  perfect_strict   \n",
       "21932                    0  perfect_strict   \n",
       "21940                    0  perfect_strict   \n",
       "21941                    0         perfect   \n",
       "21972                    0  perfect_strict   \n",
       "\n",
       "                                          ARO Categories  \n",
       "4      antibiotic inactivation; carbapenem; cephalosp...  \n",
       "5      antibiotic inactivation; carbapenem; cephalosp...  \n",
       "40                antibiotic inactivation; cephalosporin  \n",
       "45     antibiotic inactivation; carbapenem; cephalosp...  \n",
       "46     antibiotic inactivation; carbapenem; cephalosp...  \n",
       "...                                                  ...  \n",
       "21929  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "21932  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "21940  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "21941  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "21972  antibiotic target protection; tetracycline ant...  \n",
       "\n",
       "[1634 rows x 11 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prevalence[prevalence['Pathogen'].isin([\"Klebsiella pneumoniae\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "efa6295f-a552-4c2c-82bf-b733d4923b0a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ARO Accession</th>\n",
       "      <th>Name</th>\n",
       "      <th>Model ID</th>\n",
       "      <th>Model Type</th>\n",
       "      <th>Pathogen</th>\n",
       "      <th>NCBI Plasmid</th>\n",
       "      <th>NCBI WGS</th>\n",
       "      <th>NCBI Chromosome</th>\n",
       "      <th>NCBI Genomic Island</th>\n",
       "      <th>Criteria</th>\n",
       "      <th>ARO Categories</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7804</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Citrobacter koseri</td>\n",
       "      <td>12.50</td>\n",
       "      <td>3.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7805</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Citrobacter koseri</td>\n",
       "      <td>12.50</td>\n",
       "      <td>3.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7796</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella oxytoca</td>\n",
       "      <td>4.10</td>\n",
       "      <td>1.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7797</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella oxytoca</td>\n",
       "      <td>4.10</td>\n",
       "      <td>1.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7789</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>1.66</td>\n",
       "      <td>4.63</td>\n",
       "      <td>0.58</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7788</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>1.62</td>\n",
       "      <td>4.56</td>\n",
       "      <td>0.58</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7821</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella michiganensis</td>\n",
       "      <td>1.48</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7820</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella michiganensis</td>\n",
       "      <td>1.48</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7800</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Citrobacter freundii</td>\n",
       "      <td>1.28</td>\n",
       "      <td>8.10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7790</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella aerogenes</td>\n",
       "      <td>1.28</td>\n",
       "      <td>2.16</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7791</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella aerogenes</td>\n",
       "      <td>1.28</td>\n",
       "      <td>2.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7801</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Citrobacter freundii</td>\n",
       "      <td>1.28</td>\n",
       "      <td>8.33</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7798</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter cloacae</td>\n",
       "      <td>1.25</td>\n",
       "      <td>2.16</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7799</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter cloacae</td>\n",
       "      <td>1.25</td>\n",
       "      <td>2.16</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7807</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter hormaechei</td>\n",
       "      <td>0.77</td>\n",
       "      <td>1.71</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7806</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter hormaechei</td>\n",
       "      <td>0.77</td>\n",
       "      <td>1.71</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7795</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Serratia marcescens</td>\n",
       "      <td>0.66</td>\n",
       "      <td>2.90</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7794</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Serratia marcescens</td>\n",
       "      <td>0.66</td>\n",
       "      <td>2.90</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7819</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter roggenkampii</td>\n",
       "      <td>0.55</td>\n",
       "      <td>1.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7818</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter roggenkampii</td>\n",
       "      <td>0.55</td>\n",
       "      <td>1.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7787</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7786</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7784</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7785</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Salmonella enterica</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7816</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter chengduensis</td>\n",
       "      <td>0.00</td>\n",
       "      <td>4.76</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7822</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella quasipneumoniae</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.15</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7792</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Proteus mirabilis</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.27</td>\n",
       "      <td>2.20</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7793</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Proteus mirabilis</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.27</td>\n",
       "      <td>2.20</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7802</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter asburiae</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7803</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter asburiae</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7817</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter chengduensis</td>\n",
       "      <td>0.00</td>\n",
       "      <td>4.76</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7815</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Raoultella planticola</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.78</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7808</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Citrobacter werkmanii</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.56</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7814</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Raoultella planticola</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.78</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7813</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Citrobacter youngae</td>\n",
       "      <td>0.00</td>\n",
       "      <td>6.25</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7812</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Citrobacter youngae</td>\n",
       "      <td>0.00</td>\n",
       "      <td>6.25</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7811</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter kobei</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.52</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7810</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Enterobacter kobei</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.52</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7809</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Citrobacter werkmanii</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.56</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7823</th>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>1099</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella quasipneumoniae</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.15</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>perfect_strict</td>\n",
       "      <td>antibiotic inactivation; carbapenem; cephalosp...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     ARO Accession    Name  Model ID             Model Type  \\\n",
       "7804   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7805   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7796   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7797   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7789   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7788   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7821   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7820   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7800   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7790   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7791   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7801   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7798   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7799   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7807   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7806   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7795   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7794   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7819   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7818   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7787   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7786   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7784   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7785   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7816   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7822   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7792   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7793   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7802   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7803   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7817   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7815   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7808   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7814   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7813   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7812   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7811   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7810   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7809   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "7823   ARO:3001782  OXA-48      1099  protein homolog model   \n",
       "\n",
       "                        Pathogen  NCBI Plasmid  NCBI WGS  NCBI Chromosome  \\\n",
       "7804          Citrobacter koseri         12.50      3.12             0.00   \n",
       "7805          Citrobacter koseri         12.50      3.12             0.00   \n",
       "7796          Klebsiella oxytoca          4.10      1.47             0.00   \n",
       "7797          Klebsiella oxytoca          4.10      1.47             0.00   \n",
       "7789       Klebsiella pneumoniae          1.66      4.63             0.58   \n",
       "7788       Klebsiella pneumoniae          1.62      4.56             0.58   \n",
       "7821    Klebsiella michiganensis          1.48      0.60             0.00   \n",
       "7820    Klebsiella michiganensis          1.48      0.60             0.00   \n",
       "7800        Citrobacter freundii          1.28      8.10             0.00   \n",
       "7790        Klebsiella aerogenes          1.28      2.16             0.00   \n",
       "7791        Klebsiella aerogenes          1.28      2.47             0.00   \n",
       "7801        Citrobacter freundii          1.28      8.33             0.00   \n",
       "7798        Enterobacter cloacae          1.25      2.16             0.00   \n",
       "7799        Enterobacter cloacae          1.25      2.16             0.00   \n",
       "7807     Enterobacter hormaechei          0.77      1.71             0.00   \n",
       "7806     Enterobacter hormaechei          0.77      1.71             0.00   \n",
       "7795         Serratia marcescens          0.66      2.90             0.00   \n",
       "7794         Serratia marcescens          0.66      2.90             0.00   \n",
       "7819   Enterobacter roggenkampii          0.55      1.21             0.00   \n",
       "7818   Enterobacter roggenkampii          0.55      1.21             0.00   \n",
       "7787            Escherichia coli          0.24      0.41             0.90   \n",
       "7786            Escherichia coli          0.23      0.41             0.90   \n",
       "7784         Salmonella enterica          0.14      0.00             0.00   \n",
       "7785         Salmonella enterica          0.14      0.00             0.00   \n",
       "7816   Enterobacter chengduensis          0.00      4.76             0.00   \n",
       "7822  Klebsiella quasipneumoniae          0.00      1.15             0.00   \n",
       "7792           Proteus mirabilis          0.00      0.27             2.20   \n",
       "7793           Proteus mirabilis          0.00      0.27             2.20   \n",
       "7802       Enterobacter asburiae          0.00      0.89             0.00   \n",
       "7803       Enterobacter asburiae          0.00      0.89             0.00   \n",
       "7817   Enterobacter chengduensis          0.00      4.76             0.00   \n",
       "7815       Raoultella planticola          0.00      2.78             0.00   \n",
       "7808       Citrobacter werkmanii          0.00      2.56             0.00   \n",
       "7814       Raoultella planticola          0.00      2.78             0.00   \n",
       "7813         Citrobacter youngae          0.00      6.25             0.00   \n",
       "7812         Citrobacter youngae          0.00      6.25             0.00   \n",
       "7811          Enterobacter kobei          0.00      0.52             0.00   \n",
       "7810          Enterobacter kobei          0.00      0.52             0.00   \n",
       "7809       Citrobacter werkmanii          0.00      2.56             0.00   \n",
       "7823  Klebsiella quasipneumoniae          0.00      1.15             0.00   \n",
       "\n",
       "      NCBI Genomic Island        Criteria  \\\n",
       "7804                    0         perfect   \n",
       "7805                    0  perfect_strict   \n",
       "7796                    0         perfect   \n",
       "7797                    0  perfect_strict   \n",
       "7789                    0  perfect_strict   \n",
       "7788                    0         perfect   \n",
       "7821                    0  perfect_strict   \n",
       "7820                    0         perfect   \n",
       "7800                    0         perfect   \n",
       "7790                    0         perfect   \n",
       "7791                    0  perfect_strict   \n",
       "7801                    0  perfect_strict   \n",
       "7798                    0         perfect   \n",
       "7799                    0  perfect_strict   \n",
       "7807                    0  perfect_strict   \n",
       "7806                    0         perfect   \n",
       "7795                    0  perfect_strict   \n",
       "7794                    0         perfect   \n",
       "7819                    0  perfect_strict   \n",
       "7818                    0         perfect   \n",
       "7787                    0  perfect_strict   \n",
       "7786                    0         perfect   \n",
       "7784                    0         perfect   \n",
       "7785                    0  perfect_strict   \n",
       "7816                    0         perfect   \n",
       "7822                    0         perfect   \n",
       "7792                    0         perfect   \n",
       "7793                    0  perfect_strict   \n",
       "7802                    0         perfect   \n",
       "7803                    0  perfect_strict   \n",
       "7817                    0  perfect_strict   \n",
       "7815                    0  perfect_strict   \n",
       "7808                    0         perfect   \n",
       "7814                    0         perfect   \n",
       "7813                    0  perfect_strict   \n",
       "7812                    0         perfect   \n",
       "7811                    0  perfect_strict   \n",
       "7810                    0         perfect   \n",
       "7809                    0  perfect_strict   \n",
       "7823                    0  perfect_strict   \n",
       "\n",
       "                                         ARO Categories  \n",
       "7804  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7805  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7796  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7797  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7789  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7788  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7821  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7820  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7800  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7790  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7791  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7801  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7798  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7799  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7807  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7806  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7795  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7794  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7819  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7818  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7787  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7786  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7784  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7785  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7816  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7822  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7792  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7793  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7802  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7803  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7817  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7815  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7808  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7814  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7813  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7812  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7811  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7810  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7809  antibiotic inactivation; carbapenem; cephalosp...  \n",
       "7823  antibiotic inactivation; carbapenem; cephalosp...  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prevalence[prevalence['Name'].isin([\"OXA-48\"])].sort_values('NCBI Plasmid',ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "12854b6b-7d5a-4d79-9112-981bf767568c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dna_accession</th>\n",
       "      <th>pathogen</th>\n",
       "      <th>data_source</th>\n",
       "      <th>perfect_hits</th>\n",
       "      <th>strict_hits</th>\n",
       "      <th>drug_classes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>NC_009648.1</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>ncbi_chromosome</td>\n",
       "      <td>SHV-11, oqxA, Klebsiella pneumoniae KpnF, LptD</td>\n",
       "      <td>adeF, Escherichia coli marR mutant conferring ...</td>\n",
       "      <td>aminocoumarin antibiotic; aminoglycoside antib...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>NC_009650.1</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>ncbi_plasmid</td>\n",
       "      <td>TEM-1</td>\n",
       "      <td>aadA, SHV-134, AAC(6')-Ib10, OXA-896</td>\n",
       "      <td>aminoglycoside antibiotic; carbapenem; cephalo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>NC_009651.1</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>ncbi_plasmid</td>\n",
       "      <td>cmlA5, TEM-1, sul2, sul1, tet(D), ANT(2'')-Ia</td>\n",
       "      <td>catI, APH(6)-Id, APH(3'')-Ib</td>\n",
       "      <td>aminoglycoside antibiotic; cephalosporin; mono...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357</th>\n",
       "      <td>NC_012731.1</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>ncbi_chromosome</td>\n",
       "      <td>SHV-11, oqxA, Klebsiella pneumoniae KpnE, Kleb...</td>\n",
       "      <td>adeF, Escherichia coli marR mutant conferring ...</td>\n",
       "      <td>aminocoumarin antibiotic; aminoglycoside antib...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453</th>\n",
       "      <td>NC_016838.1</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>ncbi_plasmid</td>\n",
       "      <td>CTX-M-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cephalosporin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207487</th>\n",
       "      <td>NZ_WXZO01</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>SHV-1, CrcB, LptD</td>\n",
       "      <td>fosA5, adeF, Escherichia coli marR mutant conf...</td>\n",
       "      <td>aminocoumarin antibiotic; aminoglycoside antib...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207495</th>\n",
       "      <td>NZ_WXZY01</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>TEM-1, QnrS1, SHV-11, catI, OXA-48, OXA-1, CTX...</td>\n",
       "      <td>catB3, adeF, Escherichia coli marR mutant conf...</td>\n",
       "      <td>aminocoumarin antibiotic; aminoglycoside antib...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207507</th>\n",
       "      <td>NZ_WYAK01</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>TEM-1, SHV-11, sul2, QnrB1, dfrA14, OXA-1, CTX...</td>\n",
       "      <td>fosA5, catB3, adeF, Escherichia coli marR muta...</td>\n",
       "      <td>aminocoumarin antibiotic; aminoglycoside antib...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207508</th>\n",
       "      <td>NZ_WYAL01</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>TEM-1, SHV-11, catI, sul2, QnrB1, sul1, OXA-48...</td>\n",
       "      <td>fosA5, catB3, adeF, Escherichia coli marR muta...</td>\n",
       "      <td>acridine dye; aminocoumarin antibiotic; aminog...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207509</th>\n",
       "      <td>NZ_WYAM01</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>SHV-11, oqxA, Klebsiella pneumoniae KpnE, Kleb...</td>\n",
       "      <td>adeF, Escherichia coli marR mutant conferring ...</td>\n",
       "      <td>aminocoumarin antibiotic; aminoglycoside antib...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16131 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       dna_accession               pathogen      data_source  \\\n",
       "220      NC_009648.1  Klebsiella pneumoniae  ncbi_chromosome   \n",
       "221      NC_009650.1  Klebsiella pneumoniae     ncbi_plasmid   \n",
       "222      NC_009651.1  Klebsiella pneumoniae     ncbi_plasmid   \n",
       "357      NC_012731.1  Klebsiella pneumoniae  ncbi_chromosome   \n",
       "453      NC_016838.1  Klebsiella pneumoniae     ncbi_plasmid   \n",
       "...              ...                    ...              ...   \n",
       "207487     NZ_WXZO01  Klebsiella pneumoniae      ncbi_contig   \n",
       "207495     NZ_WXZY01  Klebsiella pneumoniae      ncbi_contig   \n",
       "207507     NZ_WYAK01  Klebsiella pneumoniae      ncbi_contig   \n",
       "207508     NZ_WYAL01  Klebsiella pneumoniae      ncbi_contig   \n",
       "207509     NZ_WYAM01  Klebsiella pneumoniae      ncbi_contig   \n",
       "\n",
       "                                             perfect_hits  \\\n",
       "220        SHV-11, oqxA, Klebsiella pneumoniae KpnF, LptD   \n",
       "221                                                 TEM-1   \n",
       "222         cmlA5, TEM-1, sul2, sul1, tet(D), ANT(2'')-Ia   \n",
       "357     SHV-11, oqxA, Klebsiella pneumoniae KpnE, Kleb...   \n",
       "453                                              CTX-M-14   \n",
       "...                                                   ...   \n",
       "207487                                  SHV-1, CrcB, LptD   \n",
       "207495  TEM-1, QnrS1, SHV-11, catI, OXA-48, OXA-1, CTX...   \n",
       "207507  TEM-1, SHV-11, sul2, QnrB1, dfrA14, OXA-1, CTX...   \n",
       "207508  TEM-1, SHV-11, catI, sul2, QnrB1, sul1, OXA-48...   \n",
       "207509  SHV-11, oqxA, Klebsiella pneumoniae KpnE, Kleb...   \n",
       "\n",
       "                                              strict_hits  \\\n",
       "220     adeF, Escherichia coli marR mutant conferring ...   \n",
       "221                  aadA, SHV-134, AAC(6')-Ib10, OXA-896   \n",
       "222                          catI, APH(6)-Id, APH(3'')-Ib   \n",
       "357     adeF, Escherichia coli marR mutant conferring ...   \n",
       "453                                                   NaN   \n",
       "...                                                   ...   \n",
       "207487  fosA5, adeF, Escherichia coli marR mutant conf...   \n",
       "207495  catB3, adeF, Escherichia coli marR mutant conf...   \n",
       "207507  fosA5, catB3, adeF, Escherichia coli marR muta...   \n",
       "207508  fosA5, catB3, adeF, Escherichia coli marR muta...   \n",
       "207509  adeF, Escherichia coli marR mutant conferring ...   \n",
       "\n",
       "                                             drug_classes  \n",
       "220     aminocoumarin antibiotic; aminoglycoside antib...  \n",
       "221     aminoglycoside antibiotic; carbapenem; cephalo...  \n",
       "222     aminoglycoside antibiotic; cephalosporin; mono...  \n",
       "357     aminocoumarin antibiotic; aminoglycoside antib...  \n",
       "453                                         cephalosporin  \n",
       "...                                                   ...  \n",
       "207487  aminocoumarin antibiotic; aminoglycoside antib...  \n",
       "207495  aminocoumarin antibiotic; aminoglycoside antib...  \n",
       "207507  aminocoumarin antibiotic; aminoglycoside antib...  \n",
       "207508  acridine dye; aminocoumarin antibiotic; aminog...  \n",
       "207509  aminocoumarin antibiotic; aminoglycoside antib...  \n",
       "\n",
       "[16131 rows x 6 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genomes[genomes['pathogen'].isin([\"Klebsiella pneumoniae\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "a85db781-20f0-4c0a-9816-426800bbcda7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>prevalence_sequence_id</th>\n",
       "      <th>model_id</th>\n",
       "      <th>aro_term</th>\n",
       "      <th>aro_accession</th>\n",
       "      <th>detection_model</th>\n",
       "      <th>species_name</th>\n",
       "      <th>ncbi_accession</th>\n",
       "      <th>data_type</th>\n",
       "      <th>rgi_criteria</th>\n",
       "      <th>percent_identity</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>amr_gene_family</th>\n",
       "      <th>resistance_mechanism</th>\n",
       "      <th>drug_class</th>\n",
       "      <th>card_short_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>711486</th>\n",
       "      <td>7702</td>\n",
       "      <td>1099</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Citrobacter freundii</td>\n",
       "      <td>NZ_CAJNL</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>100.00</td>\n",
       "      <td>545.8</td>\n",
       "      <td>OXA beta-lactamase</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "      <td>carbapenem; cephalosporin; penam</td>\n",
       "      <td>OXA-48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>712469</th>\n",
       "      <td>7702</td>\n",
       "      <td>1099</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>NZ_NICU01</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>100.00</td>\n",
       "      <td>545.8</td>\n",
       "      <td>OXA beta-lactamase</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "      <td>carbapenem; cephalosporin; penam</td>\n",
       "      <td>OXA-48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>712467</th>\n",
       "      <td>7702</td>\n",
       "      <td>1099</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>NZ_NICO01</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>100.00</td>\n",
       "      <td>545.8</td>\n",
       "      <td>OXA beta-lactamase</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "      <td>carbapenem; cephalosporin; penam</td>\n",
       "      <td>OXA-48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>712466</th>\n",
       "      <td>7702</td>\n",
       "      <td>1099</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>NZ_NICM01</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>100.00</td>\n",
       "      <td>545.8</td>\n",
       "      <td>OXA beta-lactamase</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "      <td>carbapenem; cephalosporin; penam</td>\n",
       "      <td>OXA-48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>712465</th>\n",
       "      <td>7702</td>\n",
       "      <td>1099</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>NZ_NICK01</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>100.00</td>\n",
       "      <td>545.8</td>\n",
       "      <td>OXA beta-lactamase</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "      <td>carbapenem; cephalosporin; penam</td>\n",
       "      <td>OXA-48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2699979</th>\n",
       "      <td>133106</td>\n",
       "      <td>1099</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>NZ_PDLW01</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>Strict</td>\n",
       "      <td>99.62</td>\n",
       "      <td>543.9</td>\n",
       "      <td>OXA beta-lactamase</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "      <td>carbapenem; cephalosporin; penam</td>\n",
       "      <td>OXA-48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2705351</th>\n",
       "      <td>135463</td>\n",
       "      <td>1099</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>NZ_SETQ01</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>Strict</td>\n",
       "      <td>99.62</td>\n",
       "      <td>543.9</td>\n",
       "      <td>OXA beta-lactamase</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "      <td>carbapenem; cephalosporin; penam</td>\n",
       "      <td>OXA-48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3372685</th>\n",
       "      <td>200878</td>\n",
       "      <td>1099</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Citrobacter freundii</td>\n",
       "      <td>NZ_JAGFN</td>\n",
       "      <td>ncbi_contig</td>\n",
       "      <td>Strict</td>\n",
       "      <td>99.62</td>\n",
       "      <td>544.7</td>\n",
       "      <td>OXA beta-lactamase</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "      <td>carbapenem; cephalosporin; penam</td>\n",
       "      <td>OXA-48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2709629</th>\n",
       "      <td>138134</td>\n",
       "      <td>1099</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>NZ_CP018712.1</td>\n",
       "      <td>ncbi_plasmid</td>\n",
       "      <td>Strict</td>\n",
       "      <td>99.55</td>\n",
       "      <td>459.1</td>\n",
       "      <td>OXA beta-lactamase</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "      <td>carbapenem; cephalosporin; penam</td>\n",
       "      <td>OXA-48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2100348</th>\n",
       "      <td>43328</td>\n",
       "      <td>1099</td>\n",
       "      <td>OXA-48</td>\n",
       "      <td>ARO:3001782</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NZ_CP045282.1</td>\n",
       "      <td>ncbi_plasmid</td>\n",
       "      <td>Strict</td>\n",
       "      <td>99.47</td>\n",
       "      <td>392.1</td>\n",
       "      <td>OXA beta-lactamase</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "      <td>carbapenem; cephalosporin; penam</td>\n",
       "      <td>OXA-48</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1469 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         prevalence_sequence_id  model_id aro_term aro_accession  \\\n",
       "711486                     7702      1099   OXA-48   ARO:3001782   \n",
       "712469                     7702      1099   OXA-48   ARO:3001782   \n",
       "712467                     7702      1099   OXA-48   ARO:3001782   \n",
       "712466                     7702      1099   OXA-48   ARO:3001782   \n",
       "712465                     7702      1099   OXA-48   ARO:3001782   \n",
       "...                         ...       ...      ...           ...   \n",
       "2699979                  133106      1099   OXA-48   ARO:3001782   \n",
       "2705351                  135463      1099   OXA-48   ARO:3001782   \n",
       "3372685                  200878      1099   OXA-48   ARO:3001782   \n",
       "2709629                  138134      1099   OXA-48   ARO:3001782   \n",
       "2100348                   43328      1099   OXA-48   ARO:3001782   \n",
       "\n",
       "               detection_model           species_name ncbi_accession  \\\n",
       "711486   protein homolog model   Citrobacter freundii       NZ_CAJNL   \n",
       "712469   protein homolog model  Klebsiella pneumoniae      NZ_NICU01   \n",
       "712467   protein homolog model  Klebsiella pneumoniae      NZ_NICO01   \n",
       "712466   protein homolog model  Klebsiella pneumoniae      NZ_NICM01   \n",
       "712465   protein homolog model  Klebsiella pneumoniae      NZ_NICK01   \n",
       "...                        ...                    ...            ...   \n",
       "2699979  protein homolog model  Klebsiella pneumoniae      NZ_PDLW01   \n",
       "2705351  protein homolog model  Klebsiella pneumoniae      NZ_SETQ01   \n",
       "3372685  protein homolog model   Citrobacter freundii       NZ_JAGFN   \n",
       "2709629  protein homolog model  Klebsiella pneumoniae  NZ_CP018712.1   \n",
       "2100348  protein homolog model       Escherichia coli  NZ_CP045282.1   \n",
       "\n",
       "            data_type rgi_criteria  percent_identity  bitscore  \\\n",
       "711486    ncbi_contig      Perfect            100.00     545.8   \n",
       "712469    ncbi_contig      Perfect            100.00     545.8   \n",
       "712467    ncbi_contig      Perfect            100.00     545.8   \n",
       "712466    ncbi_contig      Perfect            100.00     545.8   \n",
       "712465    ncbi_contig      Perfect            100.00     545.8   \n",
       "...               ...          ...               ...       ...   \n",
       "2699979   ncbi_contig       Strict             99.62     543.9   \n",
       "2705351   ncbi_contig       Strict             99.62     543.9   \n",
       "3372685   ncbi_contig       Strict             99.62     544.7   \n",
       "2709629  ncbi_plasmid       Strict             99.55     459.1   \n",
       "2100348  ncbi_plasmid       Strict             99.47     392.1   \n",
       "\n",
       "            amr_gene_family     resistance_mechanism  \\\n",
       "711486   OXA beta-lactamase  antibiotic inactivation   \n",
       "712469   OXA beta-lactamase  antibiotic inactivation   \n",
       "712467   OXA beta-lactamase  antibiotic inactivation   \n",
       "712466   OXA beta-lactamase  antibiotic inactivation   \n",
       "712465   OXA beta-lactamase  antibiotic inactivation   \n",
       "...                     ...                      ...   \n",
       "2699979  OXA beta-lactamase  antibiotic inactivation   \n",
       "2705351  OXA beta-lactamase  antibiotic inactivation   \n",
       "3372685  OXA beta-lactamase  antibiotic inactivation   \n",
       "2709629  OXA beta-lactamase  antibiotic inactivation   \n",
       "2100348  OXA beta-lactamase  antibiotic inactivation   \n",
       "\n",
       "                               drug_class card_short_name  \n",
       "711486   carbapenem; cephalosporin; penam          OXA-48  \n",
       "712469   carbapenem; cephalosporin; penam          OXA-48  \n",
       "712467   carbapenem; cephalosporin; penam          OXA-48  \n",
       "712466   carbapenem; cephalosporin; penam          OXA-48  \n",
       "712465   carbapenem; cephalosporin; penam          OXA-48  \n",
       "...                                   ...             ...  \n",
       "2699979  carbapenem; cephalosporin; penam          OXA-48  \n",
       "2705351  carbapenem; cephalosporin; penam          OXA-48  \n",
       "3372685  carbapenem; cephalosporin; penam          OXA-48  \n",
       "2709629  carbapenem; cephalosporin; penam          OXA-48  \n",
       "2100348  carbapenem; cephalosporin; penam          OXA-48  \n",
       "\n",
       "[1469 rows x 15 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index_model[index_model['card_short_name'].isin([\"OXA-48\"])].sort_values('percent_identity',ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fedb14f4-ac7b-4026-91b6-bf1b571d8893",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
